clients/src/main/java/org/apache/kafka/common/KafkaFuture.java

```diff
@@ -102,6 +110,12 @@ private void maybeComplete() {
      * Returns a new KafkaFuture that, when this future completes normally, is executed with this
      * futures's result as the argument to the supplied function.
      */
+    public abstract <R> KafkaFuture<R> thenApply(FunctionInterface<T, R> function);
+
+    /**
+     * @see KafkaFuture#thenApply(FunctionInterface)
```

Comment:
You need an `@deprecated` javadoc tag here as well as the annotation.

Reply:
Ack